src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,24 @@
+// src/hooks/useLocalStorage.ts
+
+import { useEffect, useState } from "react"
+
+export function useLocalStorage<T>(key: string, initialValue: T | (() => T)) {
+  const [value, setValue] = useState<T>(() => {
+    if (typeof window !== 'undefined') {
+        const jsonValue = localStorage.getItem(key)
+        if (jsonValue != null) return JSON.parse(jsonValue)
+    }
+
+    if (typeof initialValue === "function") {
+      return (initialValue as () => T)()
+    } else {
+      return initialValue
+    }
```

Comment:
Else can be ommited here 
```suggestion
      return initialValue
```